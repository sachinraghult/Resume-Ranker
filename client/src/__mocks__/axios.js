export default {
  get: jest.fn(() => {
    Promise.resolve({ data: {} });
    console.log("1");
  }),
};
