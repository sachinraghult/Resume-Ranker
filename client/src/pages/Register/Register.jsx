import React, { useState } from "react";
import { Link, useLocation } from "react-router-dom";
import axios from "axios";

import Google from "../../components/GoogleSignLogin/Googlereg.jsx";

import styles from "./Register.module.css";

import { Url } from "../../enums";

import { SERVER_URL } from "../../config";

// All comments are correct don't chnage
const Register = () => {
  const { state } = useLocation();

  var type;
  if (state) {
    type = state.type;
  }

  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(false);
  const [pwdError, setPwdError] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();
    setError(false);
    setPwdError(false);

    // console.log(password);
    // console.log(email);
    //quite important dont touch
    try {
      const res = await axios.post(`${SERVER_URL}/auth/register`, {
        name,
        type,
        email,
        password,
      });
      res.data && window.location.replace(Url.LOGIN);
    } catch (err) {
      setError(true);
    }
  };

  return (
    <div className={styles.loginWrapper}>
      <div className={styles.loginLeft}>
        <h3 className={styles.loginLogo}>Job Finder</h3>
        <span className={styles.loginDesc}>Enter your details...</span>
      </div>
      <div className={styles.loginRight}>
        <form className={styles.RegisterBox} onSubmit={handleSubmit}>
          <div className={styles.formtop}>
            <h2>Create Account</h2>
          </div>
          {/* <button className="googleButton" >
            <img
              width="25px"
              alt="Google sign-in"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Google_%22G%22_Logo.svg/512px-Google_%22G%22_Logo.svg.png"
            />
            Sign up with Google
          </button> */}
          <Google types={type} />
          <div className={styles.orContainer}>
            <div className={styles.line}>
              <span>
                ---------------------- or Sign up with Email
                ----------------------
              </span>
            </div>
          </div>
          <label>
            Name
            <input
              type="text"
              placeholder="Name"
              className={styles.loginInput}
              value={name}
              onChange={(e) => {
                setName(e.target.value);
              }}
              required
            />
          </label>
          <label>
            Email
            <input
              type="email"
              placeholder="Email"
              className={styles.loginInput}
              value={email}
              onChange={(e) => {
                setEmail(e.target.value);
              }}
              required
            />
          </label>
          <label>
            Password
            <input
              placeholder="Password"
              type="password"
              className={styles.loginInput}
              value={password}
              onChange={(e) => setPassword(e.target.value)}
              required
            />
          </label>
          {pwdError && <div className="">Password error</div>}
          {error && (
            <span className={styles.registerError}>Email already exists</span>
          )}
          <button className={styles.loginButton} type="submit">
            Sign up
          </button>
          <span>
            Already have an account? <Link to={Url.LOGIN}>Log in</Link>
          </span>
        </form>
      </div>
    </div>
  );
};

export default Register;
