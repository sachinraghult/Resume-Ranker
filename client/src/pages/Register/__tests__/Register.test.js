import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Register from "../Register";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";
import axios, { AxiosResponse } from "axios";

let url = "";
let body = {};

jest.mock("axios", () => ({
  ...jest.requireActual("axios"),
  get: jest.fn(),
  post: jest.fn((_url, _body) => {
    return new Promise((resolve) => {
      url = _url;
      body = _body;
      resolve(true);
    });
  }),
}));

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    )
  );
});

it("renders correctly", async () => {
  const response = {
    data: [
      { id: 1, name: "game_name1" },
      { id: 2, name: "game_name2" },
    ],
  };
  axios.post.mockImplementationOnce(() => Promise.resolve(response));
  render(
    <BrowserRouter>
      <Register />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Register />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
