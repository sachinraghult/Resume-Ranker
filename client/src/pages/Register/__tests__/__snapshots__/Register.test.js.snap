// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="loginWrapper"
>
  <div
    className="loginLeft"
  >
    <h3
      className="loginLogo"
    >
      Job Finder
    </h3>
    <span
      className="loginDesc"
    >
      Enter your details...
    </span>
  </div>
  <div
    className="loginRight"
  >
    <form
      className="RegisterBox"
      onSubmit={[Function]}
    >
      <div
        className="formtop"
      >
        <h2>
          Create Account
        </h2>
      </div>
      <div
        id="signInDiv"
      >
        Sign up
      </div>
      <div
        className="orContainer"
      >
        <div
          className="line"
        >
          <span>
            ---------------------- or Sign up with Email ----------------------
          </span>
        </div>
      </div>
      <label>
        Name
        <input
          className="loginInput"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </label>
      <label>
        Email
        <input
          className="loginInput"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </label>
      <label>
        Password
        <input
          className="loginInput"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </label>
      <button
        className="loginButton"
        type="submit"
      >
        Sign up
      </button>
      <span>
        Already have an account? 
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in
        </a>
      </span>
    </form>
  </div>
</div>
`;
